src/jit/morph.cpp

```diff
@@ -14165,28 +14165,32 @@ void                Compiler::fgMorphBlocks()
                     fgReturnCount--;
                 }
 
-                // Block is guaranteed to have last stmt as its jump kind is BBJ_RETURN.
-                // Note that it a block with jump kind BBJ_RETURN not necessarily needs to end with GT_RETURN.
+                // Note 1: A block is not guaranteed to have a last stmt if its jump kind is BBJ_RETURN.
+                // For example a method returning void could have an emtpy block with jump kind BBJ_RETURN.
+                // Such blocks do materialize as part of in-lining.
+                //
+                // Note 2: A block with jump kind BBJ_RETURN not necessarily needs to end with GT_RETURN.
```

Comment:
Grammar is awkward. Consider "Note 2: A block with jump kind BBJ_RETURN does not necessarily need to end with GT_RETURN."